src/composables/useFilter.js

```diff
@@ -0,0 +1,64 @@
+import { computed, ref } from 'vue'
+
+export const EFilterTypes = {
+  string: 0,
+  number: 1
+}
+
+export function useFilter({ data, criteria }) {
+  const filterValue = ref('')
+  const filterIsActive = ref(false)
+
+  function onFilterChange({ value, isActive }) {
```

Comment:
```suggestion
  function setFilter({ value, isActive }) {
```

I think calling this `setFilter` more clearly communicates what it does. I read the example usage first, and it took me a second to understand what `onFilterChange` did.